discovery/interface.go

```diff
@@ -95,4 +100,16 @@ type Server interface {
 // Client defines the API for Discovery Clients.
 type Client interface {
 	Search(serviceID string, query map[string]string) ([]vc.VerifiablePresentation, error)
+
+	// StartRegistration starts registration of presentations on a Discovery Service for the specified DID.
+	// Registration will be attempted immediately, and automatically refreshed.
+	// If initial registration failed, it will return ErrRegistrationFailed, but it will keep retrying.
+	// If the function is called again for the same service/DID combination, it will try to refresh the registration.
+	// It returns an error if the service or DID is invalid/unknown.
+	StartRegistration(ctx context.Context, serviceID string, subjectDID did.DID) error
```

Comment:
From an external point of view: `Register` and `Deregister`, the fact that it renews is unknown for the caller